python/ray/dashboard/client/src/pages/dashboard/Dashboard.tsx

```diff
@@ -75,20 +103,20 @@ class Dashboard extends React.Component<
 
   handleTabChange = (event: React.ChangeEvent<{}>, value: number) => {
     this.props.setTab(value);
+    if (this.tabs[value].label === "Memory") {
+      this.props.setShouldObtainMemoryTable(true);
+    } else {
+      this.props.setShouldObtainMemoryTable(false);
```

Comment:
I think we would only want to do this if we're switching away from the memory tab, right? If someone were switching back and forth between the logical and node views, for example, there would be a bunch of unnecessary requests sent to stop memory collection. This isn't a huge deal, but I do think management of this would be easier if you moved it into the MemoryTable. You could use `useEffect` within the MemoryTable component, which is hooked up to Redux, to manage starting the subscription when the memory table renders and terminating the subscription when the user switches to a different tab and the memory table resources are cleaned up.

Overall, I think you can leave the shouldObtainMemoryTable variable in Redux as is so that it can continue to be used in refreshInfo.